target_chains/cosmwasm/tools/src/pipeline.ts

```diff
@@ -0,0 +1,216 @@
+import { readFileSync, writeFileSync, existsSync, mkdirSync } from "fs";
+import { createInterface } from "readline";
+
+// This function lets you write a question to the terminal
+// And returns you the response of the user
+function readLineAsync(msg: string) {
+  const readline = createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  return new Promise((resolve) => {
+    readline.question(msg, (userRes) => {
+      resolve(userRes);
+    });
+  });
+}
+
+type StateStore<T = any> = {
+  [stepId: string]: T;
+};
+// StateManager helps in getting and setting the state locally
+// It manipulates data in-memory and once the consumer has finished manipulating it
+// They need to commit the data to permanent storage using the commit method
+class StateManager<T> {
```

Comment:
the type parameter T here seems inaccurate because the result types of the various steps are not uniform (?) IMO it would be better to drop the type parameter and have the requirement on results be something like "the result of a step must be serializable as JSON". The users of the results can cast them to a type (which is effectively what's happening now anyway).